src/main/groovy/net/vivin/gradle/versioning/tasks/TagTask.groovy

```diff
@@ -35,9 +49,24 @@ class TagTask extends DefaultTask {
         String tag = "$tagPrefix$project.version"
 
         Git git = new Git(repository)
-        def tagRef = git.tag().setAnnotated(false).setName(tag).call()
-        if(push) {
-            git.push().add(tagRef).call()
+
+        if(!tagMessage) {
+            tagRef = git.tag().setAnnotated(false).setName(tag).call()
+        } else {
+            String message = tagMessage.call()?.trim()
+            if(!message) {
+                tagRef = git.tag().setAnnotated(false).setName(tag).call()
+            } else {
+                tagRef = git.tag().setMessage(tagMessage.call()).setName(tag).call()
+            }
```

Comment:
I made this block a bit better readable, without if-elses, they are not necessary here at all.
I also slightly changed the logic to what we originally talked about, because in your current code, there is no way to create an annotated tag without a message.
I changed it to calculate the trimmed message and then if really `null` lightweight, otherwise annotated, even if the message will be empty.